src/components/apps/listing/Listing.js

```diff
@@ -301,26 +355,29 @@ function Listing({ baseId, onRouteToApp }) {
     };
 
     const handleRequestSort = (event, property) => {
-        const isAsc = orderBy === property && order === "asc";
-        setOrder(isAsc ? "desc" : "asc");
+        const isAsc =
+            orderBy === property && order === constants.SORT_ASCENDING;
+        setOrder(isAsc ? constants.SORT_DESCENDING : constants.SORT_ASCENDING);
         setOrderBy(property);
         setSelected([]);
         setPage(0);
     };
 
     const handleCategoryChange = useCallback(
-        (selectedCategory) => {
+        (category) => {
             if (
-                selectedCategory.system_id?.toLowerCase() ===
+                category.system_id?.toLowerCase() ===
                 appType.agave.toLowerCase()
             ) {
                 setFilter(null);
             }
-            setSelectedCategory(selectedCategory);
+            setCategory(category);
             setSelected([]);
-            setPage(0);
+            if (selectedPage === 0) {
+                setPage(0);
```

Comment:
Seems like having this if statement might cause issues if I'm initially browsing All Apps on page 100 and then try to view my Apps Under Development which may only have 1 page.

Reply:
Let me double check this.

Reply:
This was indeed a bug. Fixed it. Thanks.